[PM-7288] Include changes to actions in scan workflow

bwdil opened the pull request:
- Upload CheckMarx results in cx_results.sarif to GitHub
  - Arguments to support code quality scan with SonarCloud
  - Add SonarCloud to Gradle Build process to support scan

## 🎟️ Tracking

[PM-7288 ](https://bitwarden.atlassian.net/browse/PM-7288) - Deploy scanning workflow organization-wide 

## 📔 Objective

As part of our efforts to support secure code quality initiatives and find vulnerabilities through the use CheckMarx and SonarCloud, we need to implement these changes to our scan workflow. The workflows may change over time as the AppSec team further refines the process and workflow.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

None

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

bwdil committed:
[PM-7288] Include changes to actions in scan workflow
  - Upload CheckMarx results in cx_results.sarif to GitHub
  - Arguments to support code quality scan with SonarCloud
  - Add SonarCloud to Gradle Build process to support scan

withinfocus commented on build.gradle.kts:
```diff
@@ -7,4 +7,13 @@ plugins {
+        property("sonar.organization", "bitwarden")
+        property("sonar.host.url", "https://sonarcloud.io")
+    }
 }
```
⛏️ Add a newline while you're here.

bwdil replied:
Done

withinfocus commented on .github/workflows/scan.yml:
```diff
@@ -68,8 +71,8 @@ jobs:
-          # TODO: see https://docs.sonarsource.com/sonarcloud/advanced-setup/analysis-parameters/
-          # for parameters to configure
           args: >
             -Dsonar.organization=${{ github.repository_owner }}
             -Dsonar.projectKey=${{ github.repository_owner }}_${{ github.event.repository.name }}
+            -Dsonar.sources=app/src/main,app/src/debug
+            -Dsonar.tests=app/src/test,app/src/androidTest
```
🎨 Like in other repos we should perform all the configuration in one place, either here or in the Gradle config; seems like the latter is what's needed.

bwdil replied:
Done and noted

bwdil replied:
I moved the source and test paths over to the gradle build script, but I didn't see this in the sonar documentation, so hopefully they belong there. I guess it's the same as including them as arguments.

withinfocus reviewed (changes requested):
We also need the mobile team to confirm the paths included here.

bwdil committed:
move sonar sources and tests to gradle build script

bwdil committed:
add newline to scan workflow and gradle build script

bwdil committed:
Update scan.yml

bwdil committed:
Update build.gradle.kts

bwdil committed:
make changes to sonar workflow based on review in pr
  - move sonar config to build.gradle.kts under app
  - remove androidTest from sonar tests property
  - include a task to ensure sonarscan runs post build
  - declare sonar dep version info in libs.versions.toml

bwdil committed:
Merge branch 'main' into pm-7288/ghas-scan-workflow

bwdil commented on build.gradle.kts:
```diff
@@ -7,4 +7,16 @@ plugins {
     alias(libs.plugins.google.services) apply false
-}
\ No newline at end of file
+    id("org.sonarqube") version "5.0.0.4638"
```
Done and thank you for the feedback @SaintPatrck

bwdil commented on build.gradle.kts:
```diff
@@ -7,4 +7,16 @@ plugins {
+        property("sonar.sources", "app/src/main/,app/src/debug/")
+        property("sonar.tests", "app/src/test/,app/src/androidTest/")
+    }
+}
```
Done.

bwdil commented on build.gradle.kts:
```diff
@@ -7,4 +7,16 @@ plugins {
+        property("sonar.organization", "bitwarden")
+        property("sonar.host.url", "https://sonarcloud.io")
+        property("sonar.sources", "app/src/main/,app/src/debug/")
+        property("sonar.tests", "app/src/test/,app/src/androidTest/")
```
Done.

bwdil commented on build.gradle.kts:
```diff
@@ -7,4 +7,16 @@ plugins {
+    id("org.sonarqube") version "5.0.0.4638"
+}
+
+sonar {
```
Done.

bwdil committed:
Remove sonar arguments to prevent overlap with changes to gradle build

bwdil committed:
roll back sonar args to avoid errors about missing repo owner and name

bwdil commented:
@bitwarden/dept-devops can I get a review and approval?

withinfocus reviewed (approved):
Duplicative configs but it's what's needed.

bwdil committed:
Merge branch 'main' into pm-7288/ghas-scan-workflow
